docs/blueprints/EP035.rst

```diff
@@ -0,0 +1,281 @@
+        return EntityStatus.UP
+
+    Switch.register_status_func('maintenance', switch_maintenance_status)
+
+    def switch_maintenance_reason(switch):
+        if switch.id in maintenance_switches:
+            return {'maintenance'}
+        return set()
+
+    Switch.register_status_reason_func('maintenance', switch_maintenance_reason)
+
+    # Checking the status through status_reason
+
+    switch_a = Switch('Switch-A')
+    switch_b = Switch('Switch-B')
+
+    if switch_a.status_reason:
+        assert(False)
+
+    if not switch_b.status_reason:
+        assert(False)
+
+    # Filtering out specific status reasons
+
+    if switch_a.status_reason - {'maintenance'}:
+        assert(False)
+
+    if switch_b.status_reason - {'maintenance'}:
+        assert(False)
+
+
+New Kytos Events
+################
+
+The following set of event listeners are intended to be used to notify
+other NApps once a change to the status of a set of devices occurs.
+
+KytosEvent ``topology.interruptions.start``
+===========================================
+
+Notify services of an interruption starting,
+after the interruption provider has already updated the status.
+
+Expected Input:
+
+.. code-block:: python3
+
+    {
+        "type": <interruption type str>,
+        "switches": [<switch id>],
+        "interfaces": [<interface id>],
+        "links": [<link id>],
+    }
+
+Example Usage:
+
+.. code-block:: python3
+
+    # Creating events when updating status
+
+    def start_switch_maintenance(switches: set[str]):
+        affected_switches = switches - maintenance_switches
+        maintenance_switches += switches
+        controller.buffers.app.put(
+            KytosEvent(
+                name='topology.interruptions.start'
+                content={
+                    'type': 'maintenance',
+                    'switches': affected_switches,
+                }
+            )
+        )
+
+
+
+KytosEvent ``topology.interruptions.end``
+=========================================
+
+Notify services of an interruption ending,
+after the interruption provider has already updated the status.
+
+.. code-block:: python3
+
+    {
+        "type": <interruption type str>,
+        "switches": [<switch id>],
+        "interfaces": [<interface id>],
+        "links": [<link id>],
+    }
+
+Example Usage:
+
+.. code-block:: python3
+
+    # Creating events when updating status
+
+    def end_switch_maintenance(switches: set[str]):
+        affected_switches = switches & maintenance_switches
+        maintenance_switches -= switches
+        controller.buffers.app.put(
+            KytosEvent(
+                name='topology.interruptions.end'
+                content={
+                    'type': 'maintenance',
+                    'switches': affected_switches,
+    }
+            )
+        )
+
+Notes:
+
+- Upon the end of an interruption, services which had there best available paths interrupted should make an effort to reclaim interrupted paths.
```

Comment:
There might be cases where NApps might want to subscribe for `topology.interruptions.end`, but typically higher level NApps like that primarly care about links interruptions should only probably listening to `kytos/topology.[link_down|link_up]`, but `topology` should for sure be listening to `topology.interruptions.[start|end]`, that way topology can still check if `status` is UP to notify only once accordingly. Do you share the same understanding @Ktmi? That way also facilitating for `topology` to notify only once when it's actually status UP or DOWN, so for instance if one interruption has ended, but there's still another topology impacting a link then topology wouldn't even send `kytos/topology.link_up`

Reply:
Yeah, that makes sense. Continue to generate link up and down events, based on the changes induced by interruption start and end events.

Reply:
Right. Sounds good. Let's close this thread then.